lib/nodejs/NodejsStreamInputAdapter.js

```diff
@@ -0,0 +1,73 @@
+"use strict";
+
+var utils = require('../utils');
+var GenericWorker = require('../stream/GenericWorker');
+
+/**
+ * A worker that use a nodejs stream as source.
+ * @constructor
+ * @param {Readable} stream the nodejs stream.
+ */
+function NodejsStreamInputAdapter(stream) {
+    GenericWorker.call(this, "Input adapter for nodejs stream");
+    this._upstreamEnded = false;
+    this._bindStream(stream);
+}
+
+utils.inherits(NodejsStreamInputAdapter, GenericWorker);
+
+/**
+ * Prepare the stream and bind the callbacks on it.
+ * Do this ASAP on node 0.10 ! A lazy binding doesn't always work.
+ * @param {Stream} stream the nodejs stream to use.
+ */
+NodejsStreamInputAdapter.prototype._bindStream = function (stream) {
+    var self = this;
+    this._stream = stream;
+    stream.pause();
+    stream
+    .on("data", function (chunk) {
+        self.push({
+            data: chunk,
+            meta : {
```

Comment:
We don't know the total content length that will be emitted by the nodejs stream.
When generating a zip file from streams, the update function of `generateAsync()` will get abrupt changes in the total percent:

```
{ currentFile: 'dist/', percent: 0 }
{ currentFile: 'dist/jszip.js', percent: 33.333333333333336 }
[...]
{ currentFile: 'dist/jszip.js', percent: 33.333333333333336 }
{ currentFile: 'dist/jszip.min.js', percent: 66.66666666666667 }
[...]
{ currentFile: 'dist/jszip.min.js', percent: 66.66666666666667 }
{ currentFile: null, percent: 100 }
[...]
{ currentFile: null, percent: 100 }
```

We get a better "curve" if we have the full content in memory (nodejs Buffer, ArrayBuffer, etc):
```
{ currentFile: 'dist/', percent: 0 }
{ currentFile: 'dist/jszip.js', percent: 33.333333333333336 }
{ currentFile: 'dist/jszip.js', percent: 34.813265189986616 }
{ currentFile: 'dist/jszip.js', percent: 36.2931970466399 }
[...]
{ currentFile: 'dist/jszip.js', percent: 64.41190232305222 }
{ currentFile: 'dist/jszip.js', percent: 65.8918341797055 }
[...]
{ currentFile: 'dist/jszip.min.js', percent: 94.72415163097831 }
{ currentFile: 'dist/jszip.min.js', percent: 100 }
{ currentFile: null, percent: 100 }
[...]
{ currentFile: null, percent: 100 }
```
The result is not perfect with streams but it seems good enough.